apps/open_dashboard_backend/backend.cc

```diff
@@ -0,0 +1,331 @@
+
+	const QUrl qApplicationPathUrl = QUrlFromPath(qApplicationPath);
+	const QUrl qwdUrl = QUrlFromPath(qwd);
+
+	QVector<QQmlContext::PropertyPair> newProperties = {
+		{ QString("qApplication"), qApplicationPathUrl },
+		{ QString("qwd"), qwdUrl }
+	};
+
+	for (auto importPath: frontendConfig.mQmlImportPaths)
+		mEngine.addImportPath(QUrlFromPath(importPath).toString());
+
+	emit(unloadFrontendRequest());
+	mEngine.rootContext()->setContextProperties(newProperties);
+	emit(loadFrontendRequest());
+}
+
+void Backend::ReloadFrontend()
+{
+	emit(unloadFrontendRequest());
+	// This is neccesary to force a reload of the file from disk
+	mEngine.clearComponentCache();
+	emit(loadFrontendRequest());
+}
+
+void Backend::handleTimer()
+{
+	while (true)
+	{
+		const size_t BUFFER_LEN = 2048;
+		char buffer[BUFFER_LEN];
+
+		if (!mUdpImuReceiver.DataAvailable())
+			break;
+		const size_t len = mUdpImuReceiver.Receive(buffer, BUFFER_LEN);
+
+		const std::string bufferString(buffer, len);
+		std::vector<std::string> strs;
+		boost::split(strs, bufferString, boost::is_any_of(","));
+
+		for (int i=0; i<strs.size(); ++i)
+		{
+			const std::string str = boost::algorithm::trim_copy(strs[i]);
+			if (str == "81")
+			{
+				const std::string angleString = boost::algorithm::trim_copy(strs[i+3]);
+				const double angle = boost::lexical_cast<double>(angleString);
+				mVehicleDataModel.SetInputSteeringWheelAngle(angle);
+				break;
+			}
+		}
+	}
+
+	Control control;
+	VehicleDynamic vehicleDynamic;
+	Gps gps;
+	Powertrain powertrain;
+	DriverInput driverInput;
+
+	while (true)
+	{
+		OpenDashboard::Common::InboundPacket packet;
+
+		if (!mUdpReceiver.DataAvailable())
+			break;
+		const size_t len = mUdpReceiver.Receive(packet.GetData(), packet.GetBufferSize());
+
+		for (int i=0; i<packet.GetNumberOfMessages(); ++i)
+		{
+			MessageType messageType = static_cast<MessageType>(packet.GetNextMessageType());
+
+			switch (messageType)
+			{
+			case CONTROL_MSG:
+			{
+				control = packet.GetNextMessage<Control>();
+				mControlDataDynamicModel.SetState(static_cast<ControlDataDynamicModel::State>(control.state()));
+				break;
+			}
+			case VEHICLE_DYNAMIC_MSG:
+			{
+				vehicleDynamic = packet.GetNextMessage<VehicleDynamic>();
+				mVehicleDataModel.SetVehicleVelocity(vehicleDynamic.velocity());
+				mVehicleDataModel.SetVehicleAx(vehicleDynamic.ax());
+				mVehicleDataModel.SetVehicleAy(vehicleDynamic.ay());
+				mVehicleDataModel.SetVehicleAz(vehicleDynamic.az());
+				break;
+			}
+			case GPS_MSG:
+			{
+				gps = packet.GetNextMessage<Gps>();
+				mVehicleDataModel.SetPositionLon(gps.lon());
+				mVehicleDataModel.SetPositionLat(gps.lat());
+
+				mControlDataDynamicModel.SetLat(gps.lat());
+				mControlDataDynamicModel.SetLon(gps.lon());
+				break;
+			}
+			case POWERTRAIN_MSG:
+			{
+				powertrain = packet.GetNextMessage<Powertrain>();
+
+				mVehicleDataModel.SetEngineRpm(powertrain.rotation());
```

Comment:
you're passing a lot of things here, why not just have
`mVehicleDataModel.ConfigurePowerTrain(powerTrain)`